tests/EntryPoints/REST/UnapprovePageApiTest.php

```diff
@@ -20,21 +20,45 @@ class UnapprovePageApiTest extends PageApprovalsIntegrationTest {
 	use HandlerTestTrait;
 	use MockAuthorityTrait;
 
+	private InMemoryApprovalLog $approvalLog;
+
+	protected function setUp(): void {
+		parent::setUp();
+
+		$this->approvalLog = new InMemoryApprovalLog();
+	}
+
 	public function testUnapprovePageHappyPath(): void {
+		$user = $this->getTestUser();
+		$page = $this->createPageWithCategories();
+
+		$this->approvalLog->approvePage( $page->getId(), 1 );
+
 		$response = $this->executeHandler(
 			$this->newUnapprovePageApi(),
-			$this->createValidRequestData( $this->createPageWithCategories()->getRevisionRecord()->getId() ),
+			$this->createValidRequestData( $page->getId() ),
+			authority: $user->getAuthority()
 		);
 
-		$this->assertSame( 204, $response->getStatusCode() );
+		$state = $this->approvalLog->getApprovalState( $page->getId() );
+
+		$this->assertSame( 200, $response->getStatusCode() );
+		$this->assertSame(
```

Comment:
This test fails when running all tests (`make test`).
However, it does not fail when running only this class (`make test filter=UnapprovePageApiTest`) or this single test (`make test filter=testUnapprovePageHappyPath`).

I'm still trying to figure out what's happening.